javascript/javascript2/week3/main.js

```diff
@@ -0,0 +1,132 @@
+// Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+
+setTimeout(() => console.log("This string logged after 2.5 seconds."), 2500);
+
+// Create a function that takes 2 parameters: delay and stringToLog. Calling this function should log out the stringToLog after delay seconds.
+// Call the function you have created with some different arguments.
+
+function printWithDelay(delay, strToPrint) {
+  setTimeout(() => console.log(strToPrint), delay * 1000);
+}
+
+printWithDelay(4, "This string logged after 5 seconds.");
+printWithDelay(1, "This string logged after 1 seconds.");
+
+// Create a button in html. When clicking this button, use the function
+// you created in the previous task to log out the text: Called after 5 seconds 5 seconds after the button is clicked.
+
+const printBtn = document.querySelector(".print-button");
+printBtn.addEventListener("click", () =>
+  printWithDelay(5, "printed this text with 5s delay after button was clicked")
+);
+
+// Create two functions and assign them to two different variables.
+// One function logs out Earth, the other function logs out Saturn.
+// Now create a new third function that has one parameter: planetLogFunction.
+// The only thing the third function should do is call the provided parameter function.
+
+const earthLogger = () => console.log("Earth");
+const saturnLogger = () => console.log("Saturn");
+
+function planetLogFunction(func) {
+  func();
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+// Create a button with the text called "Log location".
+// When this button is clicked the location (latitude, longitude) of the user should be logged out using this browser api
+
+let currentLatitude;
+let currentLongitude;
+function logLocation() {
+  navigator.geolocation.getCurrentPosition(
+    (currentPosition) => {
```

Comment:
String interpolation would be more convenient also in here since you wouldn`t have to repeat the same code when getting the currentLatitudee and Longitude 🥷

Reply:
@Falafelqueen  here it  should be like this?
console.log(`latitude: ${currentPosition.coords.latitude} longitude: ${currentPosition.coords.longitude}`

Reply:
yes, awesome!  With the backticks, which I think you put but it just converted it to code so they dont show🦾